File: src/api/src/rendering_task.cc
#include "vkgs/rendering_task.h"

#include "vkgs/core/rendering_task.h"

namespace vkgs {

RenderingTask::RenderingTask(std::shared_ptr<core::RenderingTask> task) : task_(task) {}

RenderingTask::~RenderingTask() {}

void RenderingTask::Wait() const { task_->Wait(); }

}  // namespace vkgs

File: binding/splatstream/rendering_task.py
import numpy as np

from . import _core


class RenderingTask:
    def __init__(
        self,
        images: np.ndarray,
        shape: tuple[int],
        task: _core.RenderingTask,
    ):
        self._images = images
        self._shape = shape
        self._task = task

    def numpy(self) -> np.ndarray:
        self._task.wait()
        return self._images.reshape(*self._shape)
